src/modules/network.json

```diff
@@ -0,0 +1,467 @@
+        "default": -255,
+        "minimum": -255,
+        "maximum": 0
+      },
+      "WifiFrequency": {
+        "title": "WifiFrequency",
+        "description": "Wifi Frequency in Ghz, example 2.4Ghz and 5Ghs.",
+        "type": "number",
+        "default": 0,
+        "minimum": 0
+      },
+      "AccessPoint": {
+        "title": "AccessPoint",
+        "description": "Properties of a scanned wifi list item.",
+        "type": "object",
+        "properties": {
+          "ssid": {
+            "type": "string",
+            "description": "Name of the wifi."
+          },
+          "securityMode": {
+            "$ref": "#/components/schema/WifiSecurityMode"
+          },
+          "signalStrength": {
+            "$ref": "#/components/schema/WifiSignalStrength"
+          },
+          "frequency": {
+            "$ref": "#/components/schema/WifiFrequency"
+          }
+        }
+      },
+      "WPSSecurityPin": {
+        "title": "WPSSecurityPin",
+        "description": "Security pin type for WPS(Wifi Protected Setup).",
+        "type": "string",
+        "enum": ["PushButton", "Pin", "ManufacturerPin"]
+      },
+      "NetworkType": {
+        "title": "NetworkType",
+        "description": "Type of Network connected to the device.",
+        "type": "string",
+        "enum": ["Wifi", "Ethernet", "None"]
+      },
+      "NetworkStatus": {
+        "title": "NetworkStatus",
+        "description": "Current Network Status of the device.",
+        "type": "object",
+        "properties": {
+          "connection": {
+            "type": "boolean",
+            "description": "Connection status to the router."
+          },
+          "interface": {
+            "$ref": "#/components/schema/NetworkType"
+          },
+          "internet": {
+            "type": "boolean",
+            "description": "Connectivity to internet."
+          }
+        }
+      },
+      "WifiConnectRequest": {
+        "title": "WifiConnectRequest",
+        "description": "Current Network Status of the device.",
+        "type": "object",
+        "properties": {
+          "ssid": {
+            "schema": {
+              "type": "string"
+            }
+          },
+          "passphrase": {
+            "schema": {
+              "type": "string"
+            }
+          },
+          "securityMode": {
+            "schema": {
+              "$ref": "#/components/schema/WifiSecurityMode"
+            }
+          },
+          "timeout": {
+            "schema": {
+              "$ref": "#/components/schema/ScanTimeout"
+            }
+          }
+        }
+      },
+      "ScanTimeout": {
+        "title": "ScanTimeout",
+        "description": "Defines the timeout in seconds. If the threshold for timeout is passed without a result it will throw an error.",
+        "type": "integer",
+        "default": 0,
+        "minimum": 0,
+        "maximum": 9999
+      },
+      "ConnectTimeout": {
+        "title": "ScanTimeout",
```

Comment:
Two schemas can't have the same title, since we use it create a class/type name in various language bindings.

Does it makes sense to just have one schema here, called Timeout, so we can standardize them all in one place?

Reply:
+1 to just Timeout